### PR TITLE
Internal #9579: ENUM IN empty

### PR DESCRIPTION
**Pull request by hawkfish**
* Optimize the edge case where none of the strings are in the enum.

### CODE REVIEW

File: src/optimizer/rule/in_clause_simplification_rule.cpp [hunk 1]
```diff
@@ -125,6 +125,13 @@ unique_ptr<Expression> InEnumSimplificationRule::Apply(LogicalOperator &op, vect
 	//	then swap out the children for the valid ENUM values
 	if (in_children.size() > 1) {
 		children.swap(in_children);
+	} else if (expr.GetExpressionType() == ExpressionType::COMPARE_IN) {
+		//	IN () => false
+		return make_uniq<BoundConstantExpression>(Value::BOOLEAN(false));
```
Comment (Mytherin) [inline src/optimizer/rule/in_clause_simplification_rule.cpp]: Should this maybe be `CONSTANT_OR_NULL`? Can we add a test where the enum has `NULL` values and this is used in a `SELECT` list (so that the distinction between false / `NULL` matters)

**Review by Mytherin — COMMENTED** [review]
Thanks! Looks good - one comment

File: src/optimizer/rule/in_clause_simplification_rule.cpp (hunk 1 above)
Comment (hawkfish) [reply src/optimizer/rule/in_clause_simplification_rule.cpp]: I think we need to know if the probe has `NULL`s. So a more complex replacement is needed. Adding a row of all `NULL`s shows this.

Comment (hawkfish) [reply src/optimizer/rule/in_clause_simplification_rule.cpp]: `ISNULL(expr, NULL, false)`

Comment (Mytherin) [reply src/optimizer/rule/in_clause_simplification_rule.cpp]: That's what `CONSTANT_OR_NULL` does, no?

Comment (hawkfish) [reply src/optimizer/rule/in_clause_simplification_rule.cpp]: Not sure what that is?

```sh
(duckdb) hawkfish@Richards-MacBook-Pro-9 duckdb % git grep CONSTANT_OR_NULL
(duckdb) hawkfish@Richards-MacBook-Pro-9 duckdb % 
```

Comment (hawkfish) [reply src/optimizer/rule/in_clause_simplification_rule.cpp]: `FilterPropagateResult::FILTER_FALSE_OR_NULL`?

Comment (Mytherin) [reply src/optimizer/rule/in_clause_simplification_rule.cpp]: It's a function (`constant_or_null`) that returns either a constant or a NULL value - see `ExpressionRewriter::ConstantOrNull`. It does effectively what you're proposing (and the optimizer knows how to handle it in subsequent steps).

**Comment by Mytherin** [comment]
Thanks!
